### PR TITLE
`Summary::isEmpty()` method added

### PR DESCRIPTION
**Pull request by Smoren**
Hi @markrogoyski,

I've added new methods and covered them by tests:
* `Summary::isEmpty()`
* `Stream::isEmpty()`

### CODE REVIEW

File: tests/Summary/IsEmptyTest.php [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+        $result = Summary::isEmpty($iterable);
+
+        // Then
+        $this->assertFalse($result);
+    }
+
+    public function dataProviderForArrayFalse(): array
+    {
+        return [
+            [[1]],
```
Comment (markrogoyski) [inline tests/Summary/IsEmptyTest.php:48]: Suggestion: I recommend adding some test cases for non-empty iterables that contain only "empty" values.

For example:
```php
[
    [0]
],
[
    [[]]
],
[
    [null]
],
```

**Comment by markrogoyski** [comment]
LGTM. Just a comment about some additional test cases that clarify what empty really means.

File: tests/Summary/IsEmptyTest.php (hunk 1 above)
Comment (Smoren) [reply tests/Summary/IsEmptyTest.php:48]: Done.

**Comment by Smoren** [comment]
Thank you @markrogoyski,

I've added the new test cases you suggested.
